django_project/frontend/src/components/PdfReport/generateChartImage.tsx

```diff
@@ -0,0 +1,191 @@
+import React, { useEffect, useRef, useState } from "react";
+import { Chart, LinearScale } from "chart.js";
+import CreatePDFContent from ".";
+import { PDFDownloadLink } from "@react-pdf/renderer";
+import { Button } from "@mui/material";
+
+// import chart componenents to be rendered here
+import FirstPageCharts from "./FirstPageContent";
+
+import html2canvas from "html2canvas";
+
+Chart.register(LinearScale);
+
+const GenerateChartImages: React.FC = () => {
+
+  // define refs to capture charts
+  const firstPageRefs = useRef(null);
+
+  // define variables to save charts base64 strings
+  const [charts, setCharts] = useState([]);
+
+  // manage the preview modal
+  const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
+  const [showChartsDiv, setShowChartsDiv] = useState(false); 
+  const [loadingCharts, setLoadingCharts] = useState(true); 
+
+  useEffect(() => {
+
+    // Simulating fetching of charts data
+    setTimeout(() => {
+      setLoadingCharts(true);
+    }, 5000); 
+  }, []);
+
+  const openModal = async () => {
+    setShowChartsDiv(true);
+    setIsModalOpen(true);
+  
+    // Only fetch and generate charts if charts are not loaded yet
+    if (charts.length === 0) {
+      // Simulating fetching of charts data (e.g., from an API) with a delay
+      setTimeout(async () => {
```

Comment:
Why do we need to simulate this? Are we going to use api in the future?

Reply:
@dimasciput the charts are not directly rendered on the page so when the modal opens: the one that shows the preview ,that's when they become visible so that they can be captured as images ,so the set timeout function helps with setting the loading to true whilst the data for the charts is being fetched from the API. so when you open the modal it shows loading charts then it displays the captured images of the charts